gio-subclass/src/application.rs

```diff
@@ -0,0 +1,655 @@
+    }
+
+    fn shutdown(&self, application: &T) {
+        application.parent_shutdown();
+    }
+
+
+    // fn handle_local_options(&self, application: &T, options: &glib::VariantDict) {
+    //     application.handle_local_options(options);
+    // }
+    //
+}
+
+pub trait ApplicationImplExt<T> {}
+
+impl<S: ApplicationImpl<T>, T: ObjectType + glib::IsA<gio::Application>> ApplicationImplExt<T>
+    for S
+{
+}
+
+any_impl!(ApplicationBase, ApplicationImpl);
+
+pub unsafe trait ApplicationBase: IsA<gio::Application> + ObjectType {
+    fn parent_startup(&self) {
+        unsafe {
+            let klass = self.get_class();
+            let parent_klass = (*klass).get_parent_class() as *const gio_ffi::GApplicationClass;
+            (*parent_klass)
+                .startup
+                .map(|f| f(self.to_glib_none().0))
+                .unwrap_or(())
+        }
+    }
+
+    fn parent_activate(&self) {
+        unsafe {
+            let klass = self.get_class();
+            let parent_klass = (*klass).get_parent_class() as *const gio_ffi::GApplicationClass;
+            (*parent_klass)
+                .activate
+                .map(|f| f(self.to_glib_none().0))
+                .unwrap_or(())
+        }
+    }
+
+    fn parent_open(&self, files: &Vec<gio::File>, hint: &String) {
+        unsafe {
+            let klass = self.get_class();
+            let parent_klass = (*klass).get_parent_class() as *const gio_ffi::GApplicationClass;
+            (*parent_klass)
+                .open
+                .map(|f| {
+                    let n_files = files.len() as i32;
+                    f(
+                        self.to_glib_none().0,
+                        files.to_glib_none().0,
+                        n_files,
+                        hint.to_glib_none().0,
+                    )
+                })
+                .unwrap_or(())
+        }
+    }
+
+    fn parent_command_line(&self, cmd_line: &gio::ApplicationCommandLine) -> i32 {
+        unsafe {
+            let klass = self.get_class();
+            let parent_klass = (*klass).get_parent_class() as *const gio_ffi::GApplicationClass;
+            (*parent_klass)
+                .command_line
+                .map(|f| f(self.to_glib_none().0, cmd_line.to_glib_none().0))
+                .unwrap_or(0)
+        }
+    }
+
+    fn parent_local_command_line(&self, arguments: &mut ArgumentList) -> Option<i32> {
+        unsafe {
+            let klass = self.get_class();
+            let parent_klass = (*klass).get_parent_class() as *const gio_ffi::GApplicationClass;
+            let mut exit_status = 0;
+            let success = (*parent_klass)
+                .local_command_line
+                .map(|f| {
+                    f(
+                        self.to_glib_none().0,
+                        arguments.ptr,
+                        &mut exit_status,
+                    )
+                }).unwrap_or(glib_ffi::GFALSE);
+
```

Comment:
Here we still need to check if the parent impl has changed the argument list (i.e. removed items). We need to update our own `ArgumentList` accordingly then

Reply:
Maybe just replace the `Vec` inside the argument list with a newly created one from the pointer